back-end/validation/register.js

```diff
@@ -22,6 +25,14 @@ module.exports = function validateRegisterInput(data) {
     errors.email = "Email is invalid";
   }
 
+  // Username checks
+  if (Validator.isEmpty(data.username)) {
+    errors.username = "Username field is required";
+  }
+  if (!userRegex.test(data.username)) {
+    errors.username = "Username cannot contain spaces or special characters other than . and _ between alphanumerical characters";
```

Comment:
Similar to the password requirements, we could probably put something on the front-end to check this before it gets back here. but it's not necessarily necessary

Reply:
IMO it would be good to split the front-end validation out into a separate issue, just to avoid scope creep in this one.

Reply:
New issue created: https://github.com/hylandtechoutreach/uct-locator/issues/90